lib/workload/stateless/stacks/workflow-manager/workflow_manager/viewsets/base.py

```diff
@@ -0,0 +1,50 @@
+from abc import ABC
+from rest_framework import filters
+from django.shortcuts import get_object_or_404
+from workflow_manager.pagination import StandardResultsSetPagination
+from rest_framework.response import Response
+from rest_framework.viewsets import ReadOnlyModelViewSet
+
+
+class BaseViewSet(ReadOnlyModelViewSet, ABC):
+    lookup_value_regex = "[^/]+"  # This is to allow for special characters in the URL
+    orcabus_id_prefix = ''
+    ordering_fields = "__all__"
+    ordering = ["-orcabus_id"]
+    pagination_class = StandardResultsSetPagination
+    filter_backends = [filters.OrderingFilter, filters.SearchFilter]
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Since we have custom orcabus_id prefix for each model, we need to remove the prefix before retrieving it.
+        """
+        pk = self.kwargs.get('pk')
+        print("DEBUG: pk")
+        print(pk)
+        if pk and pk.startswith(self.orcabus_id_prefix):
+            pk = pk[len(self.orcabus_id_prefix):]
+
+        print("DEBUG: self.queryset")
+        print(self.queryset)
```

Comment:
remove print? or is it intended?

Reply:
Good point!
I cleaned up the debug/logging statements a bit.